### ISSUE TITLE
[feat] module to import appropriate .env

### ISSUE THREAD

**Issue by heaths** [Azure/azure-dev#2336]
It's great using azd to provision resources and being able to load the _.env_ files using godotenv or whatever, but it would be even easier in projects using azd this way if we provided a library to make it easier e.g.,

```go
func LoadEnv(env string) error {
    if env == "" {
        // get .azure/config@defaultEnvironment or err
    }
    // use godotenv to load environment's .env file,
    // optionally supporting fallback to/inclusion of any repo/ancestor .env file
}
```

It's not hard code to write, but any module support for azd (in Go or other languages) might help adoption.

**Comment by weikanglim** [comment]
Cleanly separated modules hasn't really been a focus of the team. In fact, it seems that the recent shift has moved us towards thinking about layers in some places.

That being said, the code to do this currently looks like something like: https://github.com/Azure/azure-dev/blob/bfc7050577348db4070f8888676ea24d0b501e03/cli/azd/cmd/util.go#L117-L127

With perhaps a substitution of `azdContext.NewContextWithDirectory(<dir>)`

**Comment by heaths** [comment]
I was actually thinking something outside of this module for use in other modules, so a log of this code may now work. Perhaps something that simply invokes `azd env get-values --output json` (optionally passing `--environment {name}`) and sets env vars similar to `godotenv` (doesn't override already-set env vars, though maybe that's optional).

**Comment by weikanglim** [comment]
@ellismg Want your advice here: is it wise at all for our tight partners to use our modules, or the version listed https://pkg.go.dev/github.com/azure/azure-dev@v0.0.0-20230525235853-f2aafb04f11b/cli/azd/pkg/environment?tab=versions? I'm not sure how our current module publishing works.

**Comment by ellismg** [comment]
To be 100% honest - I am very nervous about us partners taking dependencies on any exported methods from the azure-dev modules. 

Fundamentally, we don't think of the entire exported surface area of these modules as public.  We certainly don't intend to support the Go 1.0 promise of compatibility for all our exported members.

The artifact we ship today is the binary itself, and I view all the go code as an implementation detail of that binary.  Logically, we should view everything as under an `internal` folder.  Perhaps we should just do that at some point so folks can't accidentally break themselves into jail.  We had talked about restructuring the tree at one point and doing that, but we never wanted to deal with the churn.

I think we have enough work on our plate to build a great CLI tool and a set of experiences on top of that without bringing in the need to also produce and support a set of libraries for public adoption in addition to the application itself.

**Comment by heaths** [comment]
@ellismg said,

> and a set of experiences on top of that

That's why I proposed this. I agree it's not pri-1, but `azd` provides a great opportunity to automate workflows that are consistent with interactive use. Having a read-only module that uses azd information to facilitate that seems to fit into that vision.

I'll prototype an idea using the EXE as the API, though stabilizing the CLI and output for something like `azd env get-values (--environment {name}) --output json` would be helpful. (Executables are APIs, too, as long as they are treated as such.)

**Comment by heaths** [comment]
It's not waiting on my feedback.

**Comment by heaths** [comment]
Still want to see something done here. I've got packages in numerous languages to do this, but official support would be nice if you want an e2e ecosystem built on azd.

**Comment by jongio** [comment]
Closing — superseded by the azd exec extension (#7520 / PR #7400), which provides a first-class way to run commands with azd environment variables loaded into the shell.

**Comment by heaths** [comment]
That's not the same thing at all. You wrote `.env` files - practically an industry standard - but then require all commands - like `cargo test` - be run through `azd exec`? That's not an industry standard practice. Quite the opposite, and not easily accomplished in a myriad of environments including our own test resource provisioning scripts in Azure SDK.

I don't see why azd can't simply locate the `.env` file for the current/default environment in the root alongside `azure.yaml`, which seems to most often go in the project root where a `.env` file would be most common to find as well. This doesn't have to require any specialized code on other programs' behalf.
